### ISSUE TITLE
Yabi is only 50% width on mixed orientation displays

### ISSUE THREAD

**Issue by piskov** [amnweb/yasb#43]
I have one album and one portrait displays.

On vertical display yabi takes only 50% of screen width despite the 100% width in config. Horizontal display works as expected.

![image](https://github.com/user-attachments/assets/f381e8e9-8910-41c5-9d81-778e061096ff)

[config.txt](https://github.com/user-attachments/files/17241187/config.txt)

**Comment by amnweb** [comment]
What's happened if you set screen size instead of 100% for width?
Example `width:1920` replace 1920 with your screen size

**Comment by piskov** [comment]
That works: 1080 (I have 4k in portrait with 200% scaling).

![image](https://github.com/user-attachments/assets/202c5625-0205-408c-92ec-a1cd2b7c1a03)


… but, of course my main 6k display (with same 200% scaling) is now cropped to almost 30% ↓
![image](https://github.com/user-attachments/assets/04141db5-c97a-4a3e-8824-4390e6420d8d)

Also it would work if I set it to 200% (but, again, main display would be twice as long).

Could it be some hardcoded logic that fails on 6k screens? Or because of that mixed portrait/landscape config?

![image](https://github.com/user-attachments/assets/1a6a57eb-87bc-433e-bc17-bfb28a50b309)

1. 6016×3384 200% landscape
2. 2160×3840 200% portrait

**Comment by piskov** [comment]
But if I set both displays to landscape, 100% works as expected. So it is definitely some bug in width calculation on mixed orientations.

**Comment by piskov** [comment]
Here’s one more thing with 100% width and second display:

1. Here is portrait (takes 50%)
![image](https://github.com/user-attachments/assets/168d7e52-816a-4a59-a1f9-3b2f3ce947cd)


2. Now I switch it to landscape (full width great!):
![image](https://github.com/user-attachments/assets/8d8863ba-3072-46d9-a297-415cac422f25)


3. I switch it back to portrait — NOW it works at full width:
![image](https://github.com/user-attachments/assets/e2ff2093-332f-4bf3-97ec-9fd0742ac8d8)

4. I switch it again to album, than close and reopen yasb — it will be 50% width in *album*:
![image](https://github.com/user-attachments/assets/766ed4bb-470c-40c3-b657-c539e76fa418)

tldr; I don’t see any logic in here of what could be the culprit :-)

**Comment by piskov** [comment]
```
  status-bar:
    screens: ['DELL P2419H (1)'] 
    ...
  status-bar-2:
    screens: ['DELL P2419H (2)'] 
    ...
```

Where could I get the correct display names?

**Comment by amnweb** [comment]
> ```
>   status-bar:
>     screens: ['DELL P2419H (1)'] 
>     ...
>   status-bar-2:
>     screens: ['DELL P2419H (2)'] 
>     ...
> ```
> 
> Where could I get the correct display names?

Should be under device manager > monitors

**Comment by amnweb** [comment]
I have added options to easily identify screens.
![image](https://github.com/user-attachments/assets/9215eda9-be27-4ea4-be26-6a434983e30d)
![image](https://github.com/user-attachments/assets/c392a727-3f24-4fe7-9b7e-f67e324c2342)

**Comment by piskov** [comment]
Since you pinged me on another issue (of overlapping windows with the bar), I tried to edit the config as you suggested.

![Image](https://github.com/user-attachments/assets/cd53dad1-a765-40d2-896d-a287ae86205a)

bars:
  status-bar:
    enabled: true
    **screens: ["ProDisplayXDR"]**
    class_name: "yasb-bar"
    alignment:
      position: "top"
      center: true
    blur_effect:
      enabled: true
      acrylic: true
      dark_mode: false
      round_corners: false
      border_color: None
    window_flags:
      always_on_top: true
      windows_app_bar: true
    dimensions:
      **width: 3008** ←—————————
      height: 26
    padding:
      top: 0
      left: 0
      bottom: 0
      right: 0
    widgets:
      left: ["komorebi_workspaces", "komorebi_active_layout", "active_window"]
      center: ["clock", "apps"]
      right: [
          "media",
          # "wifi",
          #"disk",
          #"battery",
          #"language",
          # "ip_info",
          # "traffic",
          "weather",
          "cpu",
          # "github",
          "memory",
          #"wallapers",
          "volume",
          # "power_menu",
        ]
  status-bar-2:
    enabled: true
    **screens: ["LG Ultra HD"]**
    class_name: "yasb-bar"
    alignment:
      position: "top"
      center: true
    blur_effect:
      enabled: true
      acrylic: true
      dark_mode: false
      round_corners: false
      border_color: None
    window_flags:
      always_on_top: true
      windows_app_bar: true
    dimensions:
      **width: 1080** ←—————————
      height: 26
    padding:
      top: 0
      left: 0
      bottom: 0
      right: 0
    widgets:
      left: ["komorebi_workspaces", "komorebi_active_layout", "active_window"]
      center: []
      right: []


Yet for some unknown reason to me it starts the second bar from the first screen:
![Image](https://github.com/user-attachments/assets/7ba28312-1877-492f-a1e9-7ec969be7bac)


With second bar’s width set to 100% it starts from the correct position, yet takes only 50%

![Image](https://github.com/user-attachments/assets/c4598057-2a83-4cb0-b553-b129963871ce)

**Comment by amnweb** [comment]
Please close yasb, set all bar to 100% width and run `yasbc log` in terminal then start YASB and show me output.

**Comment by piskov** [comment]
![Image](https://github.com/user-attachments/assets/c1632ce1-b7aa-4f4f-aa21-0619344c08e7)

**Comment by amnweb** [comment]
> ![Image](https://github.com/user-attachments/assets/c1632ce1-b7aa-4f4f-aa21-0619344c08e7)

This is not output from new YASB, looks like you have there two version installed, please **uninstall both** and install latest one using winget `winget install --id AmN.yasb` or download installer and try again same as above.

Also read [YASB 1.5.0 Update](https://github.com/amnweb/yasb/discussions/82)

**Comment by piskov** [comment]
I got 1.3.7 from winget (it haven’t updated via winget to newer releases). And installed today 1.5.0 manually from releases today.
![Image](https://github.com/user-attachments/assets/b9640ede-e516-4c2f-9b58-a22b83a2ddb1)

That fixed the issue!

Now everything is fine whether 100% or manual width in pixels is set.

Thank you

**Comment by amnweb** [comment]
It is better to uninstall all versions and install the latest one, next time you can just use `yasbc update` command
